server/weaverbird/backends/sql_translator/steps/totals.py

```diff
@@ -0,0 +1,102 @@
+from distutils import log
+from typing import List, Sequence
+
+from weaverbird.backends.sql_translator.steps.utils.query_transformation import (
+    build_selection_query,
+)
+from weaverbird.backends.sql_translator.types import (
+    SQLPipelineTranslator,
+    SQLQuery,
+    SQLQueryDescriber,
+    SQLQueryExecutor,
+    SQLQueryRetriever,
+)
+from weaverbird.pipeline.steps import TotalsStep
+from weaverbird.pipeline.steps.aggregate import Aggregation
+from weaverbird.pipeline.steps.totals import TotalDimension
+from weaverbird.pipeline.types import ColumnName
+from weaverbird.utils.iter import combinations
+
+
+def select_total_dimensions(total_dimensions: List[TotalDimension]) -> List[str]:
+    selects = []
+    for dimension in total_dimensions:
+        selects.append(
+            f"CASE WHEN GROUPING({dimension.total_column}) = 0 THEN {dimension.total_column} ELSE '{dimension.total_rows_label}' END"
+        )
+    return selects
+
+
+def select_aggregate(aggregations: Sequence[Aggregation]) -> List[str]:
+    aggregated_cols = []
+    for aggregation in [
+        aggregation
+        for aggregation in aggregations
+        if aggregation.agg_function not in ['first', 'last']
+    ]:
+
+        for col, new_col in zip(aggregation.columns, aggregation.new_columns):
+            if aggregation.agg_function == 'count distinct':
+                aggregated_cols.append(f'COUNT(DISTINCT {col}) AS {new_col}')
+            else:
+                aggregated_cols.append(f'{aggregation.agg_function.upper()}({col}) AS {new_col}')
+    return aggregated_cols
```

Comment:
Instead of creating/using this function, it could be better to adapt the one in [utils/aggregations.py](https://github.com/ToucanToco/weaverbird/blob/master/server/weaverbird/backends/sql_translator/steps/utils/aggregation.py#L13) 
The actual function get as input an `aggregate step` and the `aggregate type`, i suggest you to adapt that function to get as input `aggregations` and `agg_type` instead !


That's mean, in line  51 of this file it should be `get_aggs_columns(step.aggregations)`, not `select_aggregate(step.aggregations)` and then adapt tests for that function too 🙏🏾  !

Reply:
they dont do the same thing. The function `get_aggs_columns` returns a list of tuple(old_column, new_column), not sure what to do with this.
This function returns a list of select string. 

It we wanted to refactor, there is the same code here:
https://github.com/ToucanToco/weaverbird/blob/69216b72fa464065c3dc2b725052a5fe61097aa8/server/weaverbird/backends/sql_translator/steps/utils/aggregation.py#L148

Reply:
ok